### PR TITLE
demo update process

### PR DESCRIPTION
**Pull request by oluademola**
# Pull Request

## What does this PR do?
Adds demo update process to the repo

## PR checklist
Please check if your PR fulfills the following requirements:
- [ ] Does this PR fix an existing issue?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: process/demo_update_ process.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+- Create a PR and request a developer advocate to review the code changes. Don’t forget to highlight breaking changes in the PR:
+    - Meilisearch API routes, as well as functions in the Meilisearch SDK that may have changed;
+    - New feature added to the demo if any;
+    - Attach the link to the full changelog to be clear about the new feature(s) and the impact on the users.
+- After the code review is approved, merge the PR(s).
+- Follow this [guide](https://github.com/meilisearch/meilisearch-migration) to update the corresponding demo server in DigitalOcean. You can find demo droplets and their URLs [here](https://www.notion.so/5de77184bb8e42f8a1baa95dec76ec33) *(private link)*.
+- Because new values for the `Default Search API Key` and `Default Admin API Key` are generated after updating Meilisearch on DigitalOcean:
+    - Update breaking changes note with the new `Default Search API Key` value;
+    - Update these changes in the code/env file;
+    - Communicate the new API keys to whoever owns the demo sandboxes.
```
Comment (CaroFG) [inline process/demo_update_ process.md:24]: It's maybe better to say "the owner of the demo sandboxes"

**Review by CaroFG — CHANGES_REQUESTED** [review]

File: process/demo_update_ process.md (hunk 1 above)
Comment (oluademola) [reply process/demo_update_ process.md]: I made the changes

**Review by CaroFG — COMMENTED** [review]
LGTM, although I'd use a numbered list instead of bullet points

**Review by CaroFG — APPROVED** [review]
LGTM

**Pull request merged**